stdlib/public/core/StringBridge.swift

```diff
@@ -623,31 +623,28 @@ extension String {
 
     _connectOrphanedFoundationSubclassesIfNeeded()
 
-    // Smol ASCII a) may bridge to tagged pointers, b) can't contain a BOM
-    if _guts.isSmallASCII {
-      let maybeTagged = _guts.asSmall.withUTF8 { bufPtr in
-        return unsafe _createCFString(
+    if _guts.isSmall {
```

Comment:
This is mostly just rearranging the code to be a bit clearer. The substantive change is is switching from grow + recursing into bridgeToObjectiveCImpl -> __StringStorage.create, which should let us create the actual size we want rather than unnecessarily padded.